src/include/miopen/cat/problem_description.hpp

```diff
@@ -68,7 +68,7 @@ struct ProblemDescription : ProblemDescriptionBase
         {
             if(xDescs[i]->GetType() != dtype)
             {
-                MIOPEN_THROW(miopenStatusBadParm, "CatForward: Tensor types do not match.");
+                return false;
```

Comment:
Same as https://github.com/ROCm/MIOpen/pull/2562#discussion_r1436579150

Reply:
I've modified it to return false in release and throw an exception with an error message in debug.

Reply:
The same `MIOPEN_THROW` should reside in the constructor. And you can remove it from here. See https://github.com/ROCm/MIOpen/pull/2562#discussion_r1437749430.

Reply:
move it to constructor